### PR TITLE
Lecture 1 solutions

### PR DESCRIPTION
**Pull request by DCode06**
### Solutions for Lecture {1}

cc @chshersh @vrom911

### CODE REVIEW

**Comment by DCode06** [comment]
Lecture 1 exercises done

**Comment by chshersh** [comment]
Hey @DCode06, I see that you already merged this PR. Would you still like to have feedback on your solutions?

**Comment by DCode06** [comment]
Yes please. I merged it by mistake. It’s my first time using GitHub. Please
provide your feedback.

Thanks!

Divya

File: src/Lecture1.hs [hunk 1]
```diff
@@ -50,7 +51,10 @@ Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
 is 25.
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-sumOfSquares x y = error "TODO!"
+sumOfSquares:: Int -> Int > Int
+sumOfSquares 0 0 = 0
```
Comment (chshersh) [inline src/Lecture1.hs:55]: You don't need to have a separate case for two zeroes here. The result will be zero even with your only line with `^`

File: src/Lecture1.hs [hunk 2]
```diff
@@ -63,8 +67,12 @@ sumOfSquares x y = error "TODO!"
 
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
-
+lastDigit::Int->Int
+lastDigit 0 = 0
```
Comment (chshersh) [inline src/Lecture1.hs:71]: This case with `0` is redundant as `rem 10` already returns the last digit for 0️⃣

File: src/Lecture1.hs [hunk 3]
```diff
@@ -63,8 +67,12 @@ sumOfSquares x y = error "TODO!"
...
+lastDigit x
+    | x < 0 = -(x `rem` 10)
+	| otherwise = x `rem` 10
```
Comment (chshersh) [inline src/Lecture1.hs:74]: That's correct! 💯 
You can simplify the implementation a bit using the standard function `abs` that returns an absolute value of a number:

```haskell
λ: abs 7
7
λ: abs (-7)
7
```

File: src/Lecture1.hs [hunk 4]
```diff
@@ -63,8 +67,12 @@ sumOfSquares x y = error "TODO!"
...
+{-lastDigit n = error "lastDigit: Not implemented!"
```
Comment (chshersh) [inline src/Lecture1.hs:75]: I see that the rest of the functions are commented due to extra open `{-` for the comment block. You can close the comment or remove this line entirely.

```suggestion
{-lastDigit n = error "lastDigit: Not implemented!"-}
```

**Review by chshersh — COMMENTED** [review]
Congrats! The first homework is complete 👏🏼 
Hope you enjoyed the course so far. Keep up the great work, and feel free to ask any questions if you have 👌🏼

I see that half of the functions are commented so tests actually don't test them.

I also noticed that GitHub Actions CI is disabled for your repository. You can enable it here to run automatic tests for your PRs:

* https://github.com/DCode06/exercises/actions

**Pull request merged**